src/utils/request-queue.js

```diff
@@ -182,7 +182,7 @@ class RequestQueue {
       if (this.options && this.options.trackingSendDelay === 0) {
         this.sendEvents();
       } else {
-        // Defer sending of events to give beforeunload time to register (avoids race condition)
```

Comment:
What happened/happens when beforeunload/visibilitychange registers?

Reply:
beforeunload/visibilitychange changes the `pageUnloading` value and if the page is unloading than we wouldn't send events and wait for the next page to send those events from the